.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,315 @@
+        version="${{ env.latest_tag }}.${{ env.commit_count }}"
+        echo "version=$version" >> $GITHUB_ENV
+        echo $version
+
+  get-system-descs:
+    needs: build-ttmlir
+    strategy:
+      fail-fast: false
+      matrix:
+        build: [
+          {runs-on: n150},
+          {runs-on: n300},
+        ]
+
+    runs-on:
+      - self-hosted
+      - ${{ matrix.build.runs-on }}
+
+    container:
+      image: ghcr.io/${{ github.repository }}/tt-mlir-ubuntu-22-04:latest
+      options: --user root --device /dev/tenstorrent/0
+      volumes:
+        - /dev/hugepages:/dev/hugepages
+        - /dev/hugepages-1G:/dev/hugepages-1G
+        - /etc/udev/rules.d:/etc/udev/rules.d
+        - /lib/modules:/lib/modules
+        - /opt/tt_metal_infra/provisioning/provisioning_env:/opt/tt_metal_infra/provisioning/provisioning_env
+
+    steps:
+    - name: Remove existing whls files
+      shell: bash
+      run: |
+        rm -f *.whl
+
+    - name: Download ttrt run whls
+      uses: actions/download-artifact@v4
+      with:
+        name: ttrt-whl-run
+
+    - name: Install ttrt run whls
+      shell: bash
+      run: |
+        rm -rf whls_env
+        python3 -m venv whls_env
+        source whls_env/bin/activate
+        pip show ttrt && pip uninstall -y ttrt
+        pip install ttrt-${{ env.version }}*.whl --force-reinstall
+        pip install torch
+        pip install loguru
+        pip install pandas
+        pip install seaborn
+        pip install graphviz
+        pip install pyyaml
+        pip install click
+
+    - name: Generate system descriptor
+      shell: bash
+      run: |
+        source whls_env/bin/activate
+        ttrt query --save-artifacts
+
+    - name: Upload system descriptor
+      uses: actions/upload-artifact@v4
+      with:
+        name: system-desc-${{ matrix.build.runs-on }}
+        path: ttrt-artifacts/system_desc.ttsys
+
+  generate-tests:
+    needs: get-system-descs
+    strategy:
+      fail-fast: false
+      matrix:
+        build: [
+          {runs-on: self-hosted, enable_perf: OFF, name: "run"},
+          {runs-on: self-hosted, enable_perf: ON, name: "perf"},
+        ]
+
+    runs-on: ${{ matrix.build.runs-on }}
+
+    container:
+      image: ghcr.io/${{ github.repository }}/tt-mlir-ubuntu-22-04:latest
+      options: --user root
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+          fetch-depth: 0
+
+    - name: Set reusable strings
+      id: strings
+      shell: bash
+      run: |
+        echo "work-dir=$(pwd)" >> "$GITHUB_OUTPUT"
+        echo "build-output-dir=$(pwd)/build" >> "$GITHUB_OUTPUT"
+
+    - name: Git safe dir
+      run: git config --global --add safe.directory ${{ steps.strings.outputs.work-dir }}
+
+    - name: Use build artifacts
+      uses: actions/download-artifact@v4
+      with:
+        name: build-artifacts-${{ matrix.build.name }}
+        path: build
+
+    - name: Config compiler
+      shell: bash
+      run: |
+        source env/activate
```

Comment:
List of compile config flags is repeating, maybe it can go into action variable
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables

Reply:
yeah, we need to consolidate the build location into one location. I'll track on a separate issue for this since its another involved task.